cabot/cabotapp/views.py

```diff
@@ -924,3 +929,174 @@ def get(self, *args, **kwargs):
+    model = Acknowledgement
+    context_object_name = 'acks'
+
+    def get_queryset(self):
+        return Acknowledgement.objects.filter(closed_at=None).order_by('status_check', '-id').prefetch_related()
+
+    def get_context_data(self, **kwargs):
+        ctx = super(AckListView, self).get_context_data(**kwargs)
+        threshold = timezone.now() - timezone.timedelta(days=7)
+        ctx['closed_acks'] = Acknowledgement.objects.filter(closed_at__gte=threshold) \
+                                                    .order_by('-closed_at')[:NUM_VISIBLE_CLOSED_ACKS]
+        return ctx
+
+
+class AckForm(GroupedModelForm):
+    class Meta(GroupedModelForm.Meta):
+        model = Acknowledgement
+        grouped_fields = (
+            ('Filter', ('status_check', 'match_if', 'note')),
+            ('Duration', ('expire_at', 'close_after_successes')),
+        )
+        widgets = {
+            'status_check': forms.Select(attrs={
+                'data-rel': 'chosen',
+                'style': 'width: 70%',
+            }),
+            'match_if': forms.RadioSelect(),
+            'note': forms.Textarea(attrs={
+                'rows': 2,
+                'style': 'width: 70%',
+            }),
+            'expire_at': TimeFromNowField(
+                times=EXPIRE_AFTER_HOURS_OPTIONS,
+                choices=[('', 'Never (dangerous!)')],
+                attrs={
+                    'data-rel': 'chosen',
+                    'style': 'width: 70%',
+                }
+            ),
+        }
+
+    def __init__(self, *args, **kwargs):
+        super(AckForm, self).__init__(*args, **kwargs)
+        self.fields['expire_at'].required = False  # expire_at can be blank
+        self.fields['close_after_successes'].required = False  # close_after_successes can be blank
+
+
+def _update_check_and_services_timeout(ack, timeout_seconds):
+    # type: (Acknowledgement, Optional[float]) -> bool
+    """
+    Runs this ack's status_check and updates the status of any associated services.
+    Since the check may take a long time to run (e.g. more than a web request), the check is run by submitting a
+    celery task and waiting up to timeout_seconds for it to complete. If it does not complete within
+    timeout_seconds, this function will return False (but the task is not aborted; it may still complete).
+
+    You should call this whenever you create or close an ack.
+    :param timeout_seconds: Timeout in seconds; if the check takes longer than this to run, will return False.
+                            If 0, will schedule the update and immediately return.
+                            If None, will block until the check completes (dangerous for UI!).
+    :return: True if the check completed within timeout_seconds, False otherwise
+    """
+    check_id = ack.status_check.id
+    start = timezone.now()
+    update_check_and_services.apply_async((check_id,))
+
+    # we don't have a result store for celery, so we just poll the DB
```

Comment:
out of scope for this PR, but do you think it'd be worth adding a result store?

Reply:
For just this case, not really. If polling the DB ever became a problem, we could dial it down or even remove it completely at the cost of slightly worse UX.